docs/concepts/services-networking/manging-hosts-file-with-host-aliases.md

```diff
@@ -0,0 +1,71 @@
+---
+assignees:
+- rickypai
+- thockin
+title: Manging Hosts File with HostAliases
+redirect_from:
+- "/docs/user-guide/manging-hosts-file-with-hostaliases/"
+- "/docs/user-guide/manging-hosts-file-with-hostaliases.html"
+---
+
+* TOC
+{:toc}
+
+kubelet [manages](https://github.com/kubernetes/kubernetes/issues/14633) the hosts file for each container of the Pod. This is to prevent Docker from [modifying](https://github.com/moby/moby/issues/17190) the file after the containers have already been started.
+
+Because of the managed-nature of the file, any user-written content will be overwritten whenever the hosts file is remounted by Kubelet in the event of a container restart or a Pod reschedule. Thus, it is not suggested to modify the contents of the file.
+
+As an alternative, Kubernetes 1.7 provides provides a `HostAliases` field for Pods to specify entries in the hosts file to provide Pod-level override of hostnames when DNS and other options are no applicable.
```

Comment:
"...Kubernetes 1.7 provides `provides` a..." -> "...Kubernetes 1.7 provides a..."

"...no applicable." -> "not applicable."